client/state/data-layer/wpcom-http/freshness-verifier.js

```diff
@@ -0,0 +1,69 @@
+export const CLEAN_OLDER_THAN = 3 * 60 * 1000;
+
+/** @type {Date} holds timestamp of the last time a cleaning operation was performed */
+let lastCleaningTime;
+
+/** @type {Map} holds request action keys, and last execution times */
+const history = new Map();
+
+/**
+ * Generate a deterministic key for comparing request actions
+ *
+ * @param   {Object} action redux action
+ * @returns {String}        action key
+ */
+const buildActionKey = ( action ) => JSON.stringify(
+	action.meta ? { ...action, meta: undefined } : action
+);
```

Comment:
this is probably too specific here. we're probably starting out only with `GET` requests because we have to be extra careful not to much around with `POST`s. further, a `GET` is fully specified by its path and query string. we should be able to build a key from a few properties. you can check in the pipeline with the dedupe request.